File: src/main/scala/committee/nova/examplemod/ExampleMod.scala
package committee.nova.examplemod

import committee.nova.examplemod.proxies.CommonProxy
import cpw.mods.fml.common.Mod.EventHandler
import cpw.mods.fml.common.event.{FMLInitializationEvent, FMLPostInitializationEvent, FMLPreInitializationEvent}
import cpw.mods.fml.common.{Mod, SidedProxy}

@Mod(modid = "examplemod", name = "ExampleMod", useMetadata = true, modLanguage = "scala")
object ExampleMod {
  final val proxyPrefix = "committee.nova.examplemod.proxies."

  @SidedProxy(serverSide = proxyPrefix + "CommonProxy", clientSide = proxyPrefix + "ClientProxy")
  var proxy: CommonProxy = _

  @EventHandler def preInit(e: FMLPreInitializationEvent): Unit = proxy.preInit(e)

  @EventHandler def init(e: FMLInitializationEvent): Unit = proxy.init(e)

  @EventHandler def postInit(e: FMLPostInitializationEvent): Unit = proxy.postInit(e)
}

File: src/main/java/committee/nova/examplemod/ExampleMod.java
package committee.nova.examplemod;

import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.Mod.EventHandler;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.client.event.RenderPlayerEvent;
import net.minecraftforge.event.ForgeSubscribe;

@Mod(modid = "examplemod", name = "ExampleMod", version = "@VERSION@", useMetadata = true)
public class ExampleMod {


	@EventHandler
	public void onPreInit(FMLPreInitializationEvent e) {
	}


	@ForgeSubscribe
	public void onRenderPlayerPost(RenderPlayerEvent.Specials.Post e) {

	}

}
